### PR TITLE
Update libmypaint, mypaint to use python 3.10

### PR DESCRIPTION
**Pull request by lukaso**
#### Description

Updates libmypaint and mypaint to use python 3.10 for builds so that there are fewer python versions floating around in macports.

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 12.6 21G115 arm64
Xcode 14.0.1 14A400

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by fhgwright** [comment]
I wondered why I was getting notified about a libmypaint PR. :-)

It's inappropriate to combine unrelated changes into a single PR.  It's also inappropriate to change an existing PR to make it a different purpose.  The new change should have been a new PR from a new branch with the new commit, referenced to master rather than the other commit.  You can still use both changes locally by making a merge branch from the two branches.  This works for any number of branches.

File: graphics/libmypaint/Portfile
Comment (reneeotten) [inline graphics/libmypaint/Portfile:26]: please don't delete this comment, instead make sure to either update the Python version on both ports as the comment tells you ;)

Comment (lukaso) [reply graphics/libmypaint/Portfile:26]: Ah, I thought the comment meant same as the upstream port, which explicitly maps out 3.8 and no further. However the build was fine so that's why I deleted the comment. I didn't realize it referred to the other port. Will make the comment more clear.

**Comment by lukaso** [comment]
> I wondered why I was getting notified about a libmypaint PR. :-)
> 
> It's inappropriate to combine unrelated changes into a single PR. It's also inappropriate to change an existing PR to make it a different purpose. The new change should have been a new PR from a new branch with the new commit, referenced to master rather than the other commit. You can still use both changes locally by making a merge branch from the two branches. This works for any number of branches.

Sorry about that. On my local master branch I had the gpsd changes so when I branched from there they came along. I immediately removed the wrong commit. This is a new branch.

**Review by mascguy — APPROVED** [review]
LGTM. @ryandesign ?

**Review by michaelld — APPROVED** [review]
LGTM even if one of my ports is no longer part of the PR :)

**Comment by ryandesign** [comment]
Are you certain all of MyPaint works correctly with Python 3.10? Do you know of an upstream statement regarding which Python versions work? If so can you provide the URL?

We had to stick with Python 3.7 for awhile because it wouldn't work with Python 3.8 initially; see https://github.com/mypaint/mypaint/issues/1087. I just want to make sure we avoid causing problems by updating the Python version.

Before this PR can be merged, the revision of MyPaint must be increased since it has library dependencies on Python modules which this PR will change.

**Comment by lukaso** [comment]
I've tested it running. And it works. I'm no mypaint expert so I can't fully test that everything works. Here is evidence of it working on 3.10 (you'll have to scroll up to see the reference to 3.10). https://github.com/mypaint/mypaint/issues/1182#issuecomment-1086028087

**Comment by reneeotten** [comment]
@lukaso you didn't increase the `revision` for the MyPaint port as requested by @ryandesign ... after that change Ryan, are you okay with this? If so, please merge when that change has been made

**Comment by reneeotten** [comment]
current Python version is 3.11 - please verify that this version works correctly and update this PR (alternatively, if it doesn't work with newer Python versions) please just close this PR

**Comment by lukaso** [comment]
Been too long on this PR for me to remember what/how I tested. Closing.

**Pull request closed without merging**
